### DISCUSSION TITLE
Can't get typescript not working on a new project

### DISCUSSION THREAD

**Discussion by stavros-k** [amannn/next-intl#972] [Q&A] [answered]
I'm setting up a new project and I can't get typescript work for translations.
I've already have a project that it works fine, which is why I'm pulling my hair out :D
Been trying to fix it for 3 days

`global.d.ts`
![image](https://github.com/amannn/next-intl/assets/47820033/eb63dd44-6e63-44c8-aa1a-717f131c8bb4)

`tsconfig.json`
![image](https://github.com/amannn/next-intl/assets/47820033/a2593a96-a741-4ffe-8b8b-f6d32bc4cd8a)

typescript version: 5.4.3

![image](https://github.com/amannn/next-intl/assets/47820033/4d65cfdc-6456-4de0-8f19-3bb90f583a90)
Only grammar autocomplete appears
![image](https://github.com/amannn/next-intl/assets/47820033/614f8fa6-7723-431c-8e0b-8a175c1ec09d)

Any ideas? I'm probably missing something stupid :D

**Comment by amannn** [comment, marked as answer]
Hey! Yep, just this week a user [reported](https://github.com/amannn/next-intl/issues/953) a better alternative to make the TypeScript integration work in this case. I've merged a docs update in https://github.com/amannn/next-intl/pull/973 that should help with your situation!

**Reply by stavros-k** [reply]
Hmm, it seems that still something isn't right (at least in my case)
![image](https://github.com/amannn/next-intl/assets/47820033/511d8fb3-0c3b-4bc5-a3b5-04962d2acd08)
![image](https://github.com/amannn/next-intl/assets/47820033/1758882c-7c64-4cb5-a776-5c9516ade2b7)

**Reply by stavros-k** [reply]
I found something...

```ts
import de from "./de.json";
import en from "./en.json";
import es from "./es.json";
import fr from "./fr.json";
import it from "./it.json";

export const translation = {
    de,
    es,
    fr,
    it,
    en,
};

type AllTranslations = typeof translation;


// Work <---
export type Messages = typeof de &
    typeof en &
    typeof es &
    typeof fr &
    typeof it;

// Does not work <---
// export type Messages = AllTranslations[keyof AllTranslations];
```

**Reply by stavros-k** [reply]
Marked as answer as it's mostly a "me issue" now :D
Just leaving it here in case someone else bumps into that issue.

Thanks!
